chore: adopt markdownlint and markdown-table-formatter for *.md

matifali opened the pull request:
Removed dependence on `prettier` in favor of `markdownlint` and `markdown-table-formatter` for markdown files.

EdwardAngert committed:
disable prettier for docs md

EdwardAngert committed:
make fmt

EdwardAngert committed:
fix md in jfrog-artifactory

Commit:
Merge remote-tracking branch 'origin/main' into 216-disable-prettier-docs-md

Commit:
add markdownlint-cli2 and rules

Commit:
make fmt

Commit:
fix typo

Commit:
alt text to contributors

EdwardAngert committed:
Apply suggestions from code review

Co-authored-by: Muhammad Atif Ali <atif@coder.com>

Commit:
move action; remove output format

Commit:
link fixes

Commit:
fixes and ignore no-bare-urls

Commit:
md fixes

Commit:
add changed.yml

Commit:
md lint changed files

Commit:
remove redundancy from docs-pr.yaml

EdwardAngert committed:
mdlinter ignore  docs/reference

EdwardAngert committed:
fix errors

matifali committed:
Merge branch 'main' into 216-disable-prettier-docs-md

matifali committed:
Integrate markdownlint and remove Prettier for docs

- Replace Prettier with markdownlint for documentation formatting.
- Rename `docs-pr.yaml` to `docs-lint.yaml` for clarity.
- Adjust `Makefile` to use `fmt/biome` instead of `fmt/prettier`.
- Remove `.prettierignore` and `.prettierignore.include` files.
- Apply markdownlint formatting across relevant documentation files.

matifali committed:
Refactor docs lint workflow in docs-lint.yaml

- Restrict triggers to main branch and pull requests
- Remove automatic fix option in markdownlint-cli2 command

matifali committed:
Update docs lint workflow to include site directory

matifali committed:
Add packageManager to site package.json

matifali committed:
Refactor actions and workflows for consistency

- Update `setup-node` action to specify pnpm version.
- Remove redundant node_modules installation command.
- Use specific commit hashes for action versions in workflows.
- Adjust paths in Makefile for better file management.

matifali committed:
Remove pnpm-lock.yaml from version control

matifali committed:
Fix markdown lint issues in docs and Makefile

- Quote glob patterns in Makefile to handle whitespace.
- Add alt text to images for better accessibility.
- Specify language for code blocks for syntax highlighting.

matifali committed:
Remove comment line handling in example parsing

JSON does not support comments, so remove the code that handled
discarding a comment line in the example parsing. This prevents
errors when reading the JSON examples.

matifali committed:
Quote markdownlint path globs in Makefile

matifali committed:
Fix formatting in provisionerGenerated.ts

Adjust the TypeScript formatting to ensure consistency in the generated code file. Remove the biome formatting step from the `gen:provisioner` script in package.json.

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 216-disable-prettier-docs-md

matifali committed:
Refactor TypeScript formatting target in Makefile

Change the fmt/biome target to fmt/ts to clarify its purpose of formatting TypeScript in the site directory.

matifali committed:
Refactor Makefile to rename fmt/ts to fmt/biome

matifali committed:
Refactor documentation for consistency in formatting

- Standardize dashboards and tables for better readability
- Use consistent markdown styles across documentation
- Align with markdownlint formatting guidelines

matifali commented:
@EdwardAngert for formatting (make markdown tables look beautiful too) we may need another tool.

EdwardAngert commented:
> @EdwardAngert for formatting (make markdown tables look beautiful too) we may need another tool.

@matifali pretty sure I can get [markdown-table-formatter](https://github.com/nvuillam/markdown-table-formatter) working in the CLI. I'll try it

matifali commented:
@aslilac I would like you to review too as I played with biome.

I am a bit confused about `biome format`, `biome check`, `biome lint`.

aslilac commented:
> I am a bit confused about `biome format`, `biome check`, `biome lint`.

because the creators seem to have made it intentionally confusing 😅

- `biome lint` is like `eslint`
- `biome format` is like `prettier`
- `biome check` does both at once _plus_ some additional stuff that isn't included in either of the others for some reason

matifali commented:
@aslilac 
> biome check does both at once plus some additional stuff that isn't included in either of the others for some reason

I scrapped through their docs and looks like `biome check` and `biome ci` are just supersets of `biome format` and `biome lint` to run both of them in dev, CI respectively. So There is no difference in features. I was just curious if we can get rid of `biome check`.
Looks like you have a similar context as me.

aslilac commented on site/biome.json:
```diff
@@ -1,6 +1,6 @@
 {
 	"files": {
-		"ignore": ["**/*Generated.ts"]
+		"ignore": ["**/*Generated.ts", "../helm/**/templates/*.yaml"]
```
this shouldn't be necessary, we only run biome from within site/

matifali replied:
I am running it on the `*.yaml` in `helm` too and want to ignore the templates folder as that is not a valid YAML.

aslilac reviewed (dismissed):
anything in a testdata/ folder needs to be reverted and ignored by formatters

aslilac commented on scripts/examplegen/main.go:
```diff
@@ -92,11 +92,6 @@ func run(lint bool) error {
 		w = io.Discard
 	}
 
-	_, err = fmt.Fprint(w, "// Code generated by examplegen. DO NOT EDIT.\n")
-	if err != nil {
-		return err
-	}
```
why are we removing this?

matifali replied:
Because the comment made the `markdownlint` cry its not a valid YAML. YAML can't have comments.

matifali replied:
We can ignore this file altogether as its a generated file.

aslilac replied:
yaml can absolutely have comments. that's an insane thing for it to insist. 😅

johnstcn replied:
Isn't this is JSON and not YAML?

matifali replied:
Oops. Thanks for correcting Cian. It was an oversight from me. For some reason `prettier` was happy with comments in JSON but `biome` is not.

I am going to remove the comment.

mafredri replied:
Please add the file to the ignore list instead of removing the comment. I don't think we should bother with formatting it.

matifali replied:
But technically isn't it wrong to have comments in `.json`? Should we instead move to `.jsonc` (in a separate PR) for such generated `json` files?

mafredri replied:
Technically, you are correct. But I don't think we should employ such strictness to generated files. If you think it's misleading, we can call it something other than `.json` too since it's a file nobody should care about other than the generator and the consumer.

I'm not much for or against `.jsonc`, I haven't seen it used anywhere before so feels weird. Even VS Code calls it's `settings.json` without the `c` and Microsoft invented it 😅

matifali replied:
Fine then. I will exclude this. Only added it as it was previously formatted using prettier (which did not complain about the comment).

mafredri replied:
@matifali maybe prettier allowed comments, but this file was always ignored by prettier 👍🏻.

matifali replied:
Yeah it was ignored in `fmt` job but was being formatted explicitly in `Makefile` just after generation. 

I will exclude it here.

mafredri replied:
@matifali where did you see that? I don't see any formatting there:

```Makefile
examples/examples.gen.json: scripts/examplegen/main.go examples/examples.go $(shell find ./examples/templates)
	go run ./scripts/examplegen/main.go > examples/examples.gen.json
```

aslilac commented on site/.markdownlint.json:
and why is this in site/ instead of the root? most of the markdown it's supposed to format has nothing to do with site/

matifali replied:
because I installed this on the site. I wanted to get red of to `packages.json`.
Also having it in the root where `package.json` lets us skip specifying the path of config with every invocation.

aslilac commented on package.json:
```diff
@@ -1,14 +1,5 @@
 	"_comment": "This version doesn't matter, it's just to allow importing from other repos.",
 	"name": "coder",
-	"version": "0.0.0",
-	"packageManager": "pnpm@9.7.1+sha512.faf344af2d6ca65c4c5c8c2224ea77a81a5e8859cbc4e06b1511ddce2f0151512431dd19e6aff31f2c6a8f5f2aced9bd2273e1fed7dd4de1868984059d2c4247",
```
don't remove this

aslilac commented on package.json:
```diff
@@ -1,14 +1,5 @@
-	"scripts": {
-		"format": "prettier --cache --write '**/*.{css,html,json,md,yaml,yml}'",
-		"format:check": "prettier --cache --check '**/*.{css,html,json,md,yaml,yml}'",
-		"storybook": "pnpm run -C site/ storybook"
```
or this

matifali replied:
why? we do not need prettier anymore. So these do not make any sense. I can keep the `storybook`.

matifali replied:
brought back.

aslilac commented on package.json:
```diff
@@ -1,14 +1,5 @@
-		"storybook": "pnpm run -C site/ storybook"
-	},
-	"devDependencies": {
-		"prettier": "3.3.3"
```
I think we should probably put the new dependency here

matifali replied:
Hmm. ok, we can move that here.

matifali committed:
Rename docs-lint.yaml to docs-ci.yaml

Reflect changes in YAML file paths and setup type.
Use markdownlint instead of files_ignore in config.
Align with formatting guidelines for docs.

matifali committed:
Discard changes to .github/actions/setup-node/action.yaml

matifali committed:
Discard changes to coderd/prometheusmetrics/testdata/agent-stats.json

matifali committed:
Discard changes to coderd/prometheusmetrics/insights/testdata/insights-metrics.json

matifali committed:
revert non-docs chnages

matifali committed:
remove prettier

matifali committed:
ignore examples.gen.json

matifali committed:
Merge branch '216-disable-prettier-docs-md' into atif/bye-bye-prettier

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
Update markdown formatting tools and apply fixes

- Remove deprecated Prettier formatting for markdown files.
- Introduce markdown-table-formatter and markdownlint-cli2 for consistent markdown formatting.
- Update workflows and Makefile to reflect changes in formatting and linting processes.
- Apply consistent table aligning across markdown documentation files for better readability.

matifali committed:
ci: specify workspace directory in setup-node action

matifali committed:
Add Storybook launch script to package.json

matifali committed:
Rename docs lint workflow to docs CI

matifali committed:
Remove redundant setup-node configuration

matifali committed:
Add echo for changed files in docs CI workflow

matifali committed:
Fix workflow environment variable syntax in docs CI

Use `\${{ }}` for accessing outputs in GitHub Actions syntax.

matifali committed:
Swap lint and fmt steps in docs workflow

Switch the order of the lint and fmt steps in the docs-ci workflow file, ensuring markdownlint-cli2 runs before markdown-table-formatter.

matifali committed:
Rename workflow job to 'docs' in docs-ci.yaml

matifali committed:
Improve markdown formatting consistency

- Add alt text to image tags in README.md for accessibility
- Convert indented numbered lists to consistent format in guide.md
- Change shell snippets to use ```shell instead of ```bash
- Harmonize headers level across various markdown documents
- Adjust tables alignment for readability
- Refactor list and header structure to improve readability and consistency

matifali committed:
Refactor JSON example tags formatting and add comment

- Added a comment to indicate the file is generated by `examplegen`.
- Improved the formatting of JSON tags by splitting each tag into its own line for clarity and consistency.

matifali committed:
Add generated examples for various environments

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
`make fmt`

matifali committed:
Resolve merge conflict by incorporating both suggestions

matifali committed:
fix glob

matifali committed:
fix glob

matifali committed:
fixup: remove extra examples.gen.json from root

matifali committed:
JSON do not have commnets

matifali committed:
remove target for ignored file

matifali committed:
make -B

matifali committed:
fixup

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
fixup

matifali committed:
Reorder and update VSCode extension recommendations

matifali committed:
enable more rules

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
make gen

matifali committed:
make fmt

matifali committed:
golden files

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
Update URLs to angle brackets for Markdown

EdwardAngert commented on .github/workflows/docs-ci.yaml:
```diff
@@ -0,0 +1,36 @@
+name: Docs CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
I think that `on>push>main` like this means it's only tracking pushes to `main`. If we replace the `branches` part with `paths` it _should_ check all pushes to all branches as long as they touch files in `docs`

```suggestion
on:
    push:
        paths:
            - "docs/**"
    pull_request:
        paths:
            - "docs/**"
```

matifali replied:
We are also linting files outside of `docs` for example README.md, CONTRIBUTING.md SECURITY.md etc. So I think the current scope is fine. Or we have to expand the `paths` scope and include all those files too which could cause a drift if we add more markdown files outside of `docs/` in the future.

matifali replied:
Addressed.

EdwardAngert commented on docs/admin/integrations/jfrog-artifactory.md:
```diff
@@ -31,84 +31,83 @@ by using our official Coder [modules](https://registry.coder.com). We publish
-Coder's [external-auth](https://coder.com/docs/admin/external-auth) feature and
-allows each user to authenticate with Artifactory using an OAuth flow and issues
-user-scoped tokens to each user.
+Coder's [external-auth](../../admin/external-auth.md) feature and allows each
```
```suggestion
Coder's [external-auth](../external-auth.md) feature and allows each
```

EdwardAngert commented on docs/admin/integrations/jfrog-artifactory.md:
```diff
@@ -31,84 +31,83 @@ by using our official Coder [modules](https://registry.coder.com). We publish
-3. Add a new
-   [external authentication](https://coder.com/docs/admin/external-auth) to
-   Coder by setting these env variables,
+1. Add a new [external authentication](../../admin/external-auth.md) to Coder by
```
```suggestion
1. Add a new [external authentication](../external-auth.md) to Coder by
```

EdwardAngert commented on docs/admin/monitoring/health-check.md:
```diff
@@ -24,15 +24,15 @@ If there is an issue, you may see one of the following errors reported:
 ### EACS01
 
-_Access URL not set_
+### Access URL not set
```
```suggestion
#### Access URL not set
```

EdwardAngert commented on docs/admin/monitoring/notifications/teams.md:
```diff
@@ -21,115 +21,115 @@ following:
-    or body content to suit your needs.
+1. Configure the Webhook Trigger.
+
+   Begin by configuring the trigger: **"When a Teams webhook request is
+   received"**.
+
+   Ensure the trigger access level is set to **"Anyone"**.
+
+1. Setup the JSON Parsing Action.
+
+   Add the **"Parse JSON"** action, linking the content to the **"Body"** of the
+   received webhook request. Use the following schema to parse the notification
```
```suggestion
   Begin by configuring the trigger: **When a Teams webhook request is
   received**.

   Ensure the trigger access level is set to **Anyone**.

1. Setup the JSON Parsing Action.

   Add the **Parse JSON** action, linking the content to the **Body** of the
   received webhook request. Use the following schema to parse the notification
```

EdwardAngert commented on docs/admin/monitoring/notifications/teams.md:
```diff
@@ -21,115 +21,115 @@ following:
+
+1. Configure the Adaptive Card Action.
+
+   Finally, set up the **"Post Adaptive Card in a chat or channel"** action with
```
```suggestion
   Finally, set up the **Post Adaptive Card in a chat or channel** action with
```

EdwardAngert commented on docs/admin/security/audit-logs.md:
```diff
@@ -8,27 +8,27 @@ We track the following resources:
-| WorkspaceBuild<br><i>start, stop</i>                     | <table><thead><tr><th>Field</th><th>Tracked</th></tr></thead><tbody><tr><td>build_number</td><td>false</td></tr><tr><td>created_at</td><td>false</td></tr><tr><td>daily_cost</td><td>false</td></tr><tr><td>deadline</td><td>false</td></tr><tr><td>id</td><td>false</td></tr><tr><td>initiator_by_avatar_url</td><td>false</td></tr><tr><td>initi ...
-| WorkspaceProxy<br><i></i>                                | <table><thead><tr><th>Field</th><th>Tracked</th></tr></thead><tbody><tr><td>created_at</td><td>true</td></tr><tr><td>deleted</td><td>false</td></tr><tr><td>derp_enabled</td><td>true</td></tr><tr><td>derp_only</td><td>true</td></tr><tr><td>display_name</td><td>true</td></tr><tr><td>icon</td><td>true</td></tr><tr><td>id</td><td>true</td>< ...
-| WorkspaceTable<br><i></i>                                | <table><thead><tr><th>Field</th><th>Tracked</th></tr></thead><tbody><tr><td>automatic_updates</td><td>true</td></tr><tr><td>autostart_schedule</td><td>true</td></tr><tr><td>created_at</td><td>false</td></tr><tr><td>deleted</td><td>false</td></tr><tr><td>deleting_at</td><td>true</td></tr><tr><td>dormant_at</td><td>true</td></tr><tr><td>f ...
+|<b>Resource<b>||
```
this `<b>` tag should close, but I'm not sure where we'd have to adjust that

matifali replied:
This is an autogenerated doc where we apply the table formatter. So I am not what could be wrong here. It does render fine.

EdwardAngert commented on docs/admin/users/idp-sync.md:
```diff
@@ -5,6 +5,8 @@
 IDP sync is an Enterprise and Premium feature.
 [Learn more](https://coder.com/pricing#compare-plans).
 
+## Group Sync
+
 </blockquote>
 
```
```suggestion
</blockquote>

## Group Sync

```

EdwardAngert commented on docs/changelogs/v2.1.1.md:
```diff
@@ -7,7 +7,7 @@
   > You can use `last_used_before` and `last_used_after` in the workspaces
   > search with [RFC3339Nano](https://www.rfc-editor.org/rfc/rfc3339) datetime
-- Add `daily_cost`` to `coder ls` to show
+- Add `daily_cost`` to`coder ls` to show
```
```suggestion
- Add `daily_cost` to `coder ls` to show
```

EdwardAngert commented on docs/changelogs/v2.5.0.md:
```diff
@@ -92,7 +92,7 @@
 
 - Align CODER_HTTP_ADDRESS with document (#10779) (@JounQin)
-- Migrate all deprecated `CODER_ADDRESS `to `CODER_HTTP_ADDRESS` (#10780) (@JounQin)
+- Migrate all deprecated `CODER_ADDRESS`to `CODER_HTTP_ADDRESS` (#10780) (@JounQin)
```
```suggestion
- Migrate all deprecated `CODER_ADDRESS` to `CODER_HTTP_ADDRESS` (#10780) (@JounQin)
```

EdwardAngert commented on docs/user-guides/workspace-access/jetbrains.md:
````diff
@@ -248,14 +248,14 @@ windows-x64, windows-aarch64, linux-x64, linux-aarch64, osx-x64, osx-aarch64
 ```
 
-**Clients**
+#### Clients
````
this should be tabs, but I'm leaving it for now since I've got a PR coming to edit the JetBrains doc

EdwardAngert reviewed (changes requested):
added some suggestions

and we shouldn't edit or lint any files in docs/reference because they're generated + will probably be overwritten

matifali commented on .github/workflows/docs-ci.yaml:
```diff
@@ -0,0 +1,36 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```
This is fine. It will get updated automatically by @dependabot. This is a requirement from OpenSSF to pin actions to commit hashes.

matifali commented on .github/workflows/docs-ci.yaml:
```diff
@@ -0,0 +1,36 @@
+      - name: Setup Node
+        uses: ./.github/actions/setup-node
+
+      - uses: tj-actions/changed-files@bab30c2299617f6615ec02a68b9a40d10bd21366 # v45.0.5
```
same here.

matifali commented on .github/workflows/docs-ci.yaml:
```diff
@@ -0,0 +1,36 @@
+      - uses: tj-actions/changed-files@bab30c2299617f6615ec02a68b9a40d10bd21366 # v45.0.5
+        id: changed-files
+        with:
+          files: "docs/**/*.md"
```
What could be other changes?

matifali replied:
And Also README.md and other *.md files outside of docs.

matifali committed:
Update codersdk/deployment.go

Co-authored-by: Edward Angert <EdwardAngert@users.noreply.github.com>

matifali commented on codersdk/deployment.go:
```diff
@@ -594,7 +594,7 @@ type ExternalAuthConfig struct {
 	//
 	// Git clone makes use of this by parsing the URL from:
-	// 'Username for "https://github.com":'
+	// 'Username for https://github.com>:'
```
Thanks. for catching this.

matifali commented on codersdk/workspaceproxy.go:
```diff
@@ -193,12 +193,12 @@ type Region struct {
 	// WildcardHostname is the wildcard hostname for subdomain apps.
-	// E.g. *.us.example.com
-	// E.g. *--suffix.au.example.com
+	// E.g. <*.us.example.com>
+	// E.g. <*--suffix.au.example.com>
```
I am not sure about this but we can try. Some of these comments get rendered as CLI help in the terminal and `` can cause it to run as a subshell command.

matifali committed:
Update docs/changelogs/v2.1.1.md

Co-authored-by: Edward Angert <EdwardAngert@users.noreply.github.com>

matifali commented on docs/admin/users/idp-sync.md:
```diff
@@ -96,7 +98,7 @@ OIDC provider will be added to the `myCoderGroupName` group in Coder.
 [azure-gids]:
 	https://github.com/MicrosoftDocs/azure-docs/issues/59766#issuecomment-664387195
```
This doesn't exist anymore.

matifali committed:
Update docs/changelogs/v2.5.0.md

Co-authored-by: Edward Angert <EdwardAngert@users.noreply.github.com>

matifali commented:
> and we shouldn't edit or lint any files in docs/reference because they're generated + will probably be overwritten

Nope they will not as I have made changes to the `Makefile` too so these files will be generated as they are now. i.e. gen + lint + fmt at the generation stage.

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
fix heading levels in IDP sync documentation

matifali committed:
Update docs CI workflow for specific file paths

Extend workflow triggers to monitor `docs/**`, markdown files, and `.github/workflows/docs-ci.yaml` for changes. Adjust changes fetching in docs lint job for alignment with new paths.

matifali committed:
Format code examples in docs with backticks

Replaced angle brackets with backticks in documentation to enhance readability and clarity, especially for example URLs and config descriptions.

matifali committed:
Update workflow paths for consistent globbing

EdwardAngert reviewed (approved):
amazing- thanks @matifali ! 

🚀

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

johnstcn reviewed (commented):
I note a number of heading changes (e.g. `#` => `##`) -- are we sure this won't break any existing documentation links?

matifali commented:
> I note a number of heading changes (e.g. # => ##) -- are we sure this won't break any existing documentation links?
There is a Markdown rule against having multiple H1 headings in a file. Yes, it will not break Markdown anchors, which are independent of heading levels. 
@EdwardAngert can confirm this.

mafredri commented on docs/reference/api/workspaces.md:
```diff
@@ -23,25 +23,25 @@ of the template will be used.
-	"template_id": "c6d67e98-83ea-49f0-8812-e4abae2b68bc",
-	"template_version_id": "0ba39c92-1f1b-4c32-aa3e-9925d7713eb1",
-	"ttl_ms": 0
+  "automatic_updates": "always",
```
Why did these indentations change from tab to space? I saw other markdown docs that retained tab indentation for JSON code blocks and tabs is what we use mostly across the code base.

matifali replied:
reverted.

mafredri commented on site/e2e/provisionerGenerated.ts:
```diff
@@ -7,1181 +7,1073 @@ export const protobufPackage = "provisioner";
-	WARN = 3,
-	ERROR = 4,
-	UNRECOGNIZED = -1,
+  TRACE = 0,
```
This file is incorrectly indented. It'll be annoying if a developer is making local changes here and the whole file is reformatted.

matifali commented on .markdownlint.json:
```diff
@@ -0,0 +1,18 @@
+{
```
Will add comments and convert to `*.jsonc`.

matifali commented on .vscode/extensions.json:
```diff
@@ -1,16 +1,16 @@
 		"github.vscode-codeql",
 		"golang.go",
 		"hashicorp.terraform",
-		"esbenp.prettier-vscode",
```
`yaml` is being handled by biome.

matifali commented on Makefile:
```diff
@@ -754,15 +776,6 @@ provisioner/terraform/testdata/version:
-	for f in $^; do
-		echo "# $${f}:" >> "$@"
-		cat "$$f" >> "$@"
-	done
```
most of the formatting is done by `biome` and it supports `.gitignore`. I need to configure it though. 
See: https://biomejs.dev/reference/configuration/#vcsuseignorefile

EdwardAngert commented:
> @johnstcn 
>> I note a number of heading changes (e.g. # => ##) -- are we sure this won't break any existing documentation links?
> 
> @matifali 
>>There is a Markdown rule against having multiple H1 headings in a file. Yes, it will not break Markdown anchors, which are independent of heading levels. @EdwardAngert can confirm this.

+1 to what @matifali says here - totally reasonable question and good catch though. Whatever the number of `#` in the markdown, they all anchor as a single `#` in the URL (which is why there's another [rule against duplicate headings](https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md) - it would just link to the first instance). If we have a table of contents on any pages, those might potentially look a little different, but it shouldn't affect anything in terms of navigation

EdwardAngert commented on .github/workflows/docs-ci.yaml:
```diff
@@ -0,0 +1,45 @@
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
+          # markdown-table-formatter requires a space separated list of files
+          echo ${{ steps.changed-files.outputs.all_changed_files }} | tr ',' '\n' | pnpm exec markdown-table-formatter --check
```
I like this - seems like it might make debugging issues later

something like a commented out debug line that writes the env contents into a file? is that possible/useful?

I'm not sure what it would specifically look like or if it's a `cat` or `echo` (`printf`?) thing, but something like:

```shell
cat <<< "$changed-files" > "list-changed-files.txt"
```

matifali replied:
@mafredri I ma leaving the suggestions here out of this PR. I need to learn the risks here and can address in a follow up PR> I also welcome you taking a jab at this.

matifali committed:
wip

matifali commented on cli/testdata/coder_server_--help.golden:
```diff
@@ -56,7 +56,7 @@ OPTIONS:
 
```
What is the best way to do that?
These files are generated from code and so does the API and CLI docs. And to make the docs follow the rules this is an artifact.

I will further explore how best to deal with this.

matifali commented on site/package.json:
```diff
@@ -20,7 +20,7 @@
 		"playwright:test": "playwright test --config=e2e/playwright.config.ts",
 		"playwright:test-ui": "playwright test --config=e2e/playwright.config.ts --ui $([[ \"$CODER\" == \"true\" ]] && echo --ui-port=7500 --ui-host=0.0.0.0)",
-		"gen:provisioner": "protoc --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=./e2e/ --ts_proto_opt=outputJsonMethods=false,outputEncodeMethods=encode-no-creation,outputClientImpl=false,nestJs=false,outputPartialMethods=false,fileSuffix=Generated,suffix=hey -I ../provisionersdk/proto ../provisionersdk/proto/provisioner.proto && pnpm exec prettier --ignore-path '/dev/null' --cache - ...
+		"gen:provisioner": "protoc --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=./e2e/ --ts_proto_opt=outputJsonMethods=false,outputEncodeMethods=encode-no-creation,outputClientImpl=false,nestJs=false,outputPartialMethods=false,fileSuffix=Generated,suffix=hey -I ../provisionersdk/proto ../provisionersdk/proto/provisioner.proto",
```
`./e2e/provisionerGenerated.ts` is a generated file so should not format I guess.
This is already in the ignore list of biome.
https://github.com/coder/coder/blob/687b015194aa0215b3ec9c7073e31c1b3087647e/site/biome.jsonc#L8

matifali commented on Makefile:
```diff
@@ -683,22 +695,32 @@ site/src/api/countriesGenerated.ts: scripts/typegen/countries.tstmpl scripts/typ
 	./scripts/pnpm_install.sh
-	pnpm exec prettier --write ./docs/reference/cli/index.md ./docs/reference/cli/*.md ./docs/manifest.json
+	pnpm exec markdownlint-cli2 --fix ./docs/reference/cli/*.md
+	pnpm exec markdown-table-formatter ./docs/reference/cli/*.md
```
Will consider in a followup

matifali committed:
wip 2

matifali commented on coderd/apidoc/docs.go:
```diff
@@ -11000,7 +11000,7 @@ const docTemplate = `{
                 },
                 "regex": {
-                    "description": "Regex allows API requesters to match an auth config by\na string (e.g. coder.com) instead of by it's type.\n\nGit clone makes use of this by parsing the URL from:\n'Username for \"https://github.com\":'\nAnd sending it to the Coder server to match against the Regex.",
+                    "description": "Regex allows API requesters to match an auth config by\na string (e.g. coder.com) instead of by it's type.\n\nGit clone makes use of this by parsing the URL from:\n'Username for \u003chttps://github.com\u003e:'\nAnd sending it to the Coder server to match against the Regex.",
```
reverted this back.

matifali committed:
fixup

matifali committed:
Update markdownlint config for schema and templates docs

Added path exclusions to resolve issues with table formatting in `api/schema.md` and `api/templates.md`.

bpmct commented:
wow

matifali committed:
Merge branch 'main' into atif/bye-bye-prettier

matifali committed:
make -B gen

matifali committed:
Delete coderd/database/dbmock/gomock_reflect_2684282611/prog.go

matifali committed:
make -B gen

matifali committed:
make -B gen

matifali committed:
make update-golden-files

matifali committed:
fixup

matifali commented on site/biome.jsonc:
```diff
@@ -6,7 +6,6 @@
 			"out/",
 			"test-results/",
 			"e2e/**/*Generated.ts",
-			"src/api/*Generated.ts",
```
removing as we want to lint/fmt these files.
